### DISCUSSION TITLE
Can not find action.yml in post step, despite successful previous step

### DISCUSSION THREAD

**Discussion by timlkelly** [community/community#25800] [Actions]
<p>I have created a composite GitHub action and placed it in a private repository so that I can reference it in multiple projects. The private repo consists only of the <code>action.yml</code> file at the top level.</p>
<p>In another repository, I am referencing this composite action with the following yaml:</p>
<pre><code class="lang-auto">jobs:
  notify_slack:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout slack notifer action
        uses: actions/checkout@v2
        with:
          repository: my-org/slack-notifier
          ref: main
          token: ${{ secrets.PAT_TOKEN }}
          path: ./.github/actions/slack-notifier

      - name: Notify Slack
        uses: ./.github/actions/slack-notifier
        with:
          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
</code></pre>
<p>When I run the action. The <code>Notify Slack</code> action succeeds, however, the <code>Post Notify Slack</code> action fails.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/d/5/d5eb38d8aa726ea2814dbebe48b3d7569e3a3673.png" data-download-href="/uploads/short-url/uwpF3fFxvmA89v58OslfgnsmshR.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/8271737/181093353-bb6e9c67-c78b-4295-95c6-ee338f2284b1.png" alt="image" data-base62-sha1="uwpF3fFxvmA89v58OslfgnsmshR" width="690" height="213" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/d/5/d5eb38d8aa726ea2814dbebe48b3d7569e3a3673_2_690x213.png, https://aws1.discourse-cdn.com/github/optimized/3X/d/5/d5eb38d8aa726ea2814dbebe48b3d7569e3a3673_2_1035x319.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/d/5/d5eb38d8aa726ea2814dbebe48b3d7569e3a3673_2_1380x426.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/5/d5eb38d8aa726ea2814dbebe48b3d7569e3a3673_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">2488×770 69 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I am do not understand what <code>action.yml</code> this error is referencing because it was able to find the <code>action.yml</code> to run the <code>Notify Slack</code> step. It does successfully post to Slack.</p>

**Comment by SergiosKolios** [comment]
<p><a class="mention" href="/timlkelly">@timlkelly</a> did you find a solution for that? I’m running into the same issue</p>

**Comment by timlkelly** [comment]
<p><a class="mention" href="/sergioskolios">@SergiosKolios</a> not yet.</p>

**Comment by SergiosKolios** [comment]
<p>I solved it, in my case it was because I was invoking checkout actions from my custom action, in your case that would be <code>slack-notifier</code>.</p>

**Comment by timlkelly** [comment]
<p>Ah, I see. Yes, I am also performing a checkout action of the project inside of the custom action.</p>
<p><a class="mention" href="/sergioskolios">@SergiosKolios</a> did you solve this by removing the checkout action inside the custom action? or was there a different solution?</p>

**Comment by timlkelly** [comment, marked as answer]
<p>I fixed this by specifying the <code>path</code> parameter in my custom action</p>
<pre><code class="lang-yaml">runs:
  using: composite
  steps:
    - name: Checkout
      uses: actions/checkout@v2
      with:
        path: ${{ github.repository }}
        fetch-depth: 0
</code></pre>

**Comment by SergiosKolios** [comment]
<p>Hi <a class="mention" href="/timlkelly">@timlkelly</a>, yes that should do the job, otherwise it’ll checkout the entire repo and do a cleanup before in the home directory of the instance. Thus it’ll fail to cleanup what was checked out before invoking the action, at least that’s how I understand it. Good luck</p>
